routes/weatherProviders/DarkSky.ts

```diff
@@ -25,35 +25,37 @@ async function getDarkSkyWateringData( coordinates: GeoCoordinates ): Promise< W
 		return undefined;
 	}
 
+	// The number of hourly forecasts to use from today's data. This will only include elements that contain historic
+	// data (not forecast data).
+	// Find the first element that contains forecast data.
+	const todayElements = Math.min( 24, todayData.hourly.data.findIndex( ( data ) => data.time > todayTimestamp - 60 * 60 ) );
+
+	const samples = [
+		// Take as much data as possible from the first elements of today's data.
+		...todayData.hourly.data.slice( 0, todayElements ),
+		// Take the remaining data from the last elements of yesterday's data.
+		...yesterdayData.hourly.data.slice( todayElements - 24 ),
+	];
```

Comment:
Suggest swap L35 and L37 so that `samples` is an ordered list (i.e. samples = [...yesterday, ...today ] ).

That way L58 is always `raining = samples[-1].precipIntensity > 0;` and no need to calculate `mostRecentSample`.